integration-tests/rpc_integration_test.go

```diff
@@ -0,0 +1,465 @@
+//                   can't meaningfully sanity check the returnedSnapshotID in
+//                   this test. Unit testing should be implemented to verify that
+//                   this logic is correct, if necessary.
+func TestGetOrdersSuccess(t *testing.T) {
+	teardownSubTest := setupSubTest(t)
+	defer teardownSubTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	// NOTE(jalextowle): Cancel in case execution stops before other cancel is called below.
+	defer cancel()
+
+	// Remove the old database and p2p files.
+	removeOldFiles(t, ctx)
+
+	buildStandaloneForTests(t, ctx)
+
+	// logMessages is a channel through which log messages from the
+	// node will be sent. We use a large buffer so it doesn't cause
+	// goroutines to block.
+	logMessages := make(chan string, 1024)
+
+	ethClient := ethclient.NewClient(ethRPCClient)
+	signedTestOrder := scenario.CreateZRXForWETHSignedTestOrder(t, ethClient, makerAddress, takerAddress, wethAmount, zrxAmount)
+	expectedOrderHash, err := signedTestOrder.ComputeOrderHash()
+	require.NoError(t, err, "could not compute order hash for standalone order")
+
+	expectedFillableTakerAssetAmount := signedTestOrder.TakerAssetAmount
+
+	expectedPage := 0
+	expectedPerPage := 5
+	expectedSnapshotID := ""
+
+	// Set up the dummy handler with an addOrdersHandler
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+
+	count := int(atomic.AddInt32(nodeCount, 1))
+
+	go func() {
+		defer wg.Done()
+		startStandaloneNode(t, ctx, count, logMessages)
+	}()
+
+	_, err = waitForLogSubstring(ctx, logMessages, "started RPC server")
+	require.NoError(t, err, "RPC server didn't start")
+
+	client, err := rpc.NewClient(standaloneRPCEndpoint + strconv.Itoa(rpcPort+count))
+	require.NoError(t, err)
+
+	signedTestOrders := []*zeroex.SignedOrder{signedTestOrder}
+	validationResponse, err := client.AddOrders(signedTestOrders)
+	require.NoError(t, err)
+	assert.Len(t, validationResponse.Accepted, 1)
+	assert.Len(t, validationResponse.Rejected, 0)
+
+	getOrdersResponse, err := client.GetOrders(expectedPage, expectedPerPage, expectedSnapshotID)
+	require.NoError(t, err)
+	expectedOrderHash, err = signedTestOrder.ComputeOrderHash()
+	require.NoError(t, err)
+	assert.Len(t, getOrdersResponse.OrdersInfos, 1)
+
+	// We need to call ResetHash so that unexported hash field is equal in later
+	// assertions.
+	signedTestOrder.ResetHash()
+
+	orderInfo := getOrdersResponse.OrdersInfos[0]
+	assert.Equal(t, expectedOrderHash, orderInfo.OrderHash, "orderHashes did not match")
+	assert.Equal(t, signedTestOrder, orderInfo.SignedOrder, "signedOrder did not match")
+	assert.Equal(t, expectedFillableTakerAssetAmount, orderInfo.FillableTakerAssetAmount, "fillableTakerAssetAmount did not match")
+
+	// Cancel the context and wait for all outstanding goroutines to finish.
+	cancel()
+	wg.Wait()
+}
+
+func TestAddPeer(t *testing.T) {
+	t.Skip("The AddPeer test is currently skipped because of nondeterministic behavior that causes it to intermittently fail")
```

Comment:
Just a heads up. I'm not reviewing this test since it's skipped and not a high priority.